bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/ForecastSolarObject.java

```diff
@@ -157,8 +158,11 @@ public double getActualEnergyValue(ZonedDateTime queryDateTime) throws SolarFore
     @Override
     public TimeSeries getEnergyTimeSeries(QueryMode mode) {
         TimeSeries ts = new TimeSeries(Policy.REPLACE);
+        Instant now = Instant.now(Utils.getClock());
         wattHourMap.forEach((timestamp, energy) -> {
-            ts.add(timestamp.toInstant(), Utils.getEnergyState(energy / 1000.0));
+            if (timestamp.toInstant().isAfter(now)) {
```

Comment:
You could store `timestamp.toInstant()` in a local variable to avoid converting it twice.